### PR TITLE
7/7

### PR DESCRIPTION
**Pull request by Abdulla2022**
Added APIManger class to request the restaurants data from yelp

### CODE REVIEW

File: Vegetarian-Food-Finder/Api/ApiManger.swift [hunk 1]
```diff
@@ -2,8 +2,35 @@
...
 import UIKit
 
+final class ApiManger {
+    
+    private class ApiConstant {
+        static let ApiUrl = "https://api.yelp.com/v3/businesses/search?location=SF&categories=vegetarian"
+        static let apiKey = "_fWbIvhY5n5FFDT2iiUnJ2BKwupqLzTbqd7kFco46l6cioKrn2-VnnIU_97bEEJG5Tp-XeHsdcSYWTXV0tO4vFTzq61DbeLrbcA1d0g6wjQ46AzI2gnV3iuCvtHFYnYx"
+    }
+    
+    func makeBusinessRequest() {
```
Comment (gustavopolitano) [inline Vegetarian-Food-Finder/Api/ApiManger.swift:17]: naming: I know that Yelp endpoint contains `business` in the URL. But try to name this method closer to what it actually does. Something like: `searchBusinesses`. You're not returning a request - it's used internally but not returned in the response, so I wouldn't include it in the method name.

File: Vegetarian-Food-Finder/Api/ApiManger.swift [hunk 2]
```diff
@@ -2,8 +2,35 @@
 //  ApiManger.swift
 //  Vegetarian-Food-Finder
 //
-//  Created by Abdullahi Ahmed on 7/6/22.
+//  Created by Abdullahi Ahmed on 7/7/22.
 //
 
 import UIKit
 
+final class ApiManger {
```
Comment (gustavopolitano) [inline Vegetarian-Food-Finder/Api/ApiManger.swift:10]: This Api is handling Yelp requests only. Let's name it closer to what it does: for example `YelpManager`. Also there's a typo in `Manger`.

File: Vegetarian-Food-Finder/Api/ApiManger.swift [hunk 3]
```diff
@@ -2,8 +2,35 @@
...
+    
+    private class ApiConstant {
```
Comment (gustavopolitano) [inline Vegetarian-Food-Finder/Api/ApiManger.swift:12]: if the constants are being used internally only, you don't need to wrap it in another private class. Just declare the constants directly into `ApiManager`.

File: Vegetarian-Food-Finder/Api/ApiManger.swift [hunk 4]
```diff
@@ -2,8 +2,35 @@
...
+        static let ApiUrl = "https://api.yelp.com/v3/businesses/search?location=SF&categories=vegetarian"
```
Comment (gustavopolitano) [inline Vegetarian-Food-Finder/Api/ApiManger.swift:13]: this URL is for searching a business, which will apply to only one of the methods of this API. I'd split it into the domain + version part ( `https://api.yelp.com/v3/`) and append the rest inside the method. That way you can reuse the same domain in multiple requests later.

**Review by gustavopolitano — APPROVED** [review]
Hey, overall looks good.

Make sure you do format the files before committing. This is such a small thing to do and is important for code quality.

I'd also think more about naming things. Remember to always aim to make it easier for future readers of the code.

One big ticket item here is the chance of learning and using async/await. It'll actually make it easier for you to write async code.

File: Vegetarian-Food-Finder/Model/Restaurant.swift [hunk 5]
```diff
@@ -0,0 +1,18 @@
+//
+//  Restaurant.swift
+//  Vegetarian-Food-Finder
+//
+//  Created by Abdullahi Ahmed on 7/7/22.
+//
+
+import UIKit
+
+ final class Resturant: NSObject {
```
[the remark this answers is not in the document]
Comment (gustavopolitano) [reply Vegetarian-Food-Finder/Model/Restaurant.swift:10]: Another example: https://www.swiftbysundell.com/basics/codable/

File: Vegetarian-Food-Finder/Api/ApiManger.swift [hunk 6]
```diff
@@ -2,8 +2,35 @@
...
+        static let apiKey = "_fWbIvhY5n5FFDT2iiUnJ2BKwupqLzTbqd7kFco46l6cioKrn2-VnnIU_97bEEJG5Tp-XeHsdcSYWTXV0tO4vFTzq61DbeLrbcA1d0g6wjQ46AzI2gnV3iuCvtHFYnYx"
+    }
+    
+    func makeBusinessRequest() {
+        let apikey = ApiConstant.apiKey
+        let url = URL(string: ApiConstant.ApiUrl)
+        var request = URLRequest(url: url!)
+        request.setValue("Bearer \(apikey)", forHTTPHeaderField: "Authorization")
+        request.httpMethod = "GET"
+        URLSession.shared.dataTask(with: request) { (data, response, error) in
```
[the remark this answers is not in the document]
Comment (gustavopolitano) [reply Vegetarian-Food-Finder/Api/ApiManger.swift:23]: let response = await URLSession.shared.data(with: request)
// now you can use response

File: Vegetarian-Food-Finder/Api/ApiManger.swift (hunk 1 above)
Comment (gustavopolitano) [inline Vegetarian-Food-Finder/Api/ApiManger.swift:17]:
```
func async searchBusiness(query: string) throws -> [Business] { 
```

**Comment by Abdulla2022** [comment]
done

**Pull request closed without merging**